### DISCUSSION TITLE
onCreateNode Mdx node missing fields

### DISCUSSION THREAD

**Discussion by dustinlacewell** [gatsbyjs/gatsby#31745] [Help] [answered]
I am using `onCreateNode` to handle `Mdx` type nodes and I'm finding that the node data is missing some expected fields. When I `console.table(Object.keys(node))` I get the following:

```
┌─────────┬────────────────────┐
│ (index) │       Values       │
├─────────┼────────────────────┤
│    0    │        'id'        │
│    1    │     'children'     │
│    2    │      'parent'      │
│    3    │     'internal'     │
│    4    │   'frontmatter'    │
│    5    │     'excerpt'      │
│    6    │     'exports'      │
│    7    │     'rawBody'      │
│    8    │ 'fileAbsolutePath' │
└─────────┴────────────────────┘
```

Where is `body`? Where is `slug`, etc?

**Comment by LekoArts** [comment, marked as answer]
This is expected behavior as it adds those fields during the `createSchemaCustomization` step which runs after the `createNodes` (when `onCreateNode` is called).

https://github.com/gatsbyjs/gatsby/blob/a67d339ff87ad0db5a519594ad1ecc8afd1c5be7/packages/gatsby-plugin-mdx/gatsby/create-schema-customization.js#L121-L294

If you want to access them/do something with them you'll have to use the `createSchemaCustomization`/`createResolvers` API.
